dbms/src/Functions/FunctionsJson.h

```diff
@@ -530,11 +612,723 @@ class FunctionsJsonArray : public IFunction
+                    {
+                        if (tmp_buf.size() < size_t_flen)
+                            tmp_buf.resize(size_t_flen);
+                        std::memcpy(tmp_buf.data(), slice.data, slice.size);
+                        std::fill(tmp_buf.data() + slice.size, tmp_buf.data() + size_t_flen, 0);
+                        JsonBinary::appendOpaque(
+                            write_buffer,
+                            JsonBinary::Opaque{from_type_code, StringRef{tmp_buf.data(), size_t_flen}});
+                    }
+                }
+            }
+            else
+            {
+                JsonBinary::appendOpaque(
+                    write_buffer,
+                    JsonBinary::Opaque{from_type_code, StringRef{slice.data, slice.size}});
+            }
+            writeChar(0, write_buffer);
+            offsets_to[i] = write_buffer.count();
+            data_from->next();
+        }
+    }
+
+    template <bool is_nullable>
+    static void doExecuteForParsingJson(
+        ColumnString::Chars_t & data_to,
+        ColumnString::Offsets & offsets_to,
+        const std::unique_ptr<IStringSource> & data_from,
+        const NullMap & null_map_from,
+        size_t size)
+    {
+        // json_type + string end char 0 + value
+        size_t reserve_size = (size * (1 + 1)) + data_from->getSizeForReserve();
+        JsonBinary::JsonBinaryWriteBuffer write_buffer(data_to, reserve_size);
+        simdjson::dom::parser parser;
+        for (size_t i = 0; i < size; ++i)
+        {
+            if constexpr (is_nullable)
+            {
+                if (null_map_from[i])
+                {
+                    writeChar(0, write_buffer);
+                    offsets_to[i] = write_buffer.count();
+                    data_from->next();
+                    continue;
+                }
+            }
+
+            const auto & slice = data_from->getWhole();
+            if (unlikely(slice.size == 0))
+                throw Exception("Invalid JSON text: The document is empty.");
+
+            const auto & json_elem = parser.parse(slice.data, slice.size);
+            if (unlikely(json_elem.error()))
+            {
+                throw Exception(fmt::format(
+                    "Invalid JSON text: The document root must not be followed by other values, details: {}",
+                    simdjson::error_message(json_elem.error())));
+            }
+            JsonBinary::appendSIMDJsonElem(write_buffer, json_elem.value_unsafe());
+
+            writeChar(0, write_buffer);
+            offsets_to[i] = write_buffer.count();
+            data_from->next();
+        }
+    }
+
+    static void doExecuteForOthers(
+        ColumnString::Chars_t & data_to,
+        ColumnString::Offsets & offsets_to,
+        const std::unique_ptr<IStringSource> & data_from,
+        size_t size)
+    {
+        // json_type + from_type_code + string end char 0
+        size_t reserve_size = (size * (1 + 1)) + data_from->getSizeForReserve();
```

Comment:
Ditto, and there is not `from_type_code` in this case?

Reply:
ok, updated.